### ISSUE TITLE
AMDGPU/GlobalISel: call in graphics shader with too many input SGPRs

### ISSUE THREAD

**Issue by petar-avramovic** [llvm/llvm-project#119376]
`llc -march=amdgcn -mcpu=gfx1010 -global-isel test.ll -o -` or `gfx900` fail with `LLVM ERROR: call in graphics shader with too many input SGPRs`
`gfx1100` works fine
cat test.ll
```
declare amdgpu_gfx void @g()

define amdgpu_ps void @f(i32 inreg %a0, i32 inreg %a1, i32 inreg %a2, i32 inreg %a3, i32 inreg %a4, i32 inreg %a5, i32 inreg %a6, i32 inreg %a7, i32 inreg %a8, i32 inreg %a9, i32 inreg %a10, i32 inreg %a11, i32 inreg %a12, i32 inreg %a13, i32 inreg %a14, i32 inreg %a15, i32 inreg %a16, i32 inreg %a17, i32 inreg %a18, i32 inreg %a19, i32 inreg %a20, i32 inreg %a21, i32 inreg %a22, i32 inreg %a23, i32 inreg %a24, i32 inreg %a25, i32 inreg %a26, i32 inreg %a27, i32 inreg %a28, i32 inreg %a29, i32 inreg %a30, i32 inreg %a31) {
bb:
  call amdgpu_gfx void @g()
  unreachable
}
```

**Comment by arsenm** [comment]
We ought to handle excess SGPR arguments by packing VGPRs with writelane, which may then further end up in memory

**Comment by jayfoad** [comment]
> We ought to handle excess SGPR arguments by packing VGPRs with writelane, which may then further end up in memory

No, this is for the amdgpu_ps calling convention. We need so support passing in values in the first ~ 35 SGPRs, since that is what the hardware will do when it launches one.

(For GFX11 and GFX12 it looks like the max number of SGPRs passed in is 35, for older architectures it is a few more than that.)

**Comment by arsenm** [comment]
> > We ought to handle excess SGPR arguments by packing VGPRs with writelane, which may then further end up in memory
> 
> No, this is for the amdgpu_ps calling convention. We need so support passing in values in the first ~ 35 SGPRs, since that is what the hardware will do when it launches one.
> 

We still do need to implement this for the callable functions though.
